lib/hdt.js

```diff
@@ -64,21 +71,21 @@ HdtDocumentPrototype.searchTerms = function (options) {
 
   // Return predicates that connect subject and object
   if (subject && object) {
-    return this.searchTriples(subject, undefined, object, { limit: limit })
+    return this.searchTriples(subject, null, object, { limit: limit })
       .then(result => result.triples.map(statement => statement.predicate));
   }
-
+  const dataFactory = this.dataFactory;
   // Return distinct terms
   return new Promise((resolve, reject) => {
     if ('subject' in options || 'object' in options) {
       if (!subject && !object) return resolve([]);
-      this._fetchDistinctTerms(subject, object, limit, posId,
-        (error, results) => error ? reject(error) : resolve(results));
+      this._fetchDistinctTerms(termToString(subject) || '', termToString(object) || '', limit, posId,
```

Comment:
termToString is guarenteed to return null or undefined if it cannot parse? @rubensworks ?

Reply:
It will return null if the term is falsy: https://github.com/rubensworks/rdf-string.js/blob/master/lib/TermUtil.ts#L23-L25

Reply:
Then we are good to go!